pylint/config/config_file_parser.py

```diff
@@ -94,6 +94,10 @@ def parse_config_file(
         if self.verbose_mode:
             print(f"Using config file {file_path}", file=sys.stderr)
 
-        if file_path.suffix == ".toml":
-            return self._parse_toml_file(file_path)
-        return self._parse_ini_file(file_path)
+        try:
+            if file_path.suffix == ".toml":
+                return self._parse_toml_file(file_path)
+            return self._parse_ini_file(file_path)
+        except (configparser.Error, tomllib.TOMLDecodeError) as e:
+            self.linter.add_message("config-parse-error", line=0, args=str(e))
+            return {}, []
```

Comment:
I think this should be handled inside ``_parse_toml_file`` and ``_parse_ini_file``?

Reply:
I think the intent is clearer this way: for any config failure (other than a missing pylint section in toml), raise this error.

There would also be a lot more basically duplicate try catch blocks if we make them more granular.